rules-tests/DowngradePhp70/Rector/TryCatch/DowngradeCatchThrowableRector/Fixture/fixture.php.inc

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Rector\Tests\DowngradePhp70\Rector\TryCatch\DowngradeCatchThrowableRector\Fixture;
+
+try {
+    // Some code...
+} catch (\Throwable $e) {
+    // Code to handle the exception
+} catch (AnotherException $e) {
+    // ...
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\DowngradePhp70\Rector\TryCatch\DowngradeCatchThrowableRector\Fixture;
+
+try {
+    // Some code...
+} catch (\Throwable|\Exception $e) {
```

Comment:
This code won't work on php 5.6, ref https://3v4l.org/dZY6t#v5.6.40

Reply:
I am aware, that’s why I wrote it should be used with `DowngradePipeToMultiCatchExceptionRector` in the `RuleDefinition`.

Reply:
That seems invalid way as the code is not exists in "before" not work in php 7.0 too, make it piped will introduce bug

Reply:
If we use it in conjunction with `Rector\DowngradePhp71\Rector\TryCatch\DowngradePipeToMultiCatchExceptionRector`, it will work on PHP 7.0. The `DowngradePhp71` ruleset should be part of `DowngradePhp70` so it should work unless one is invoking this specific rule manually.

The alternative would be duplicating the `DowngradePipeToMultiCatchExceptionRector` within this rule.

Reply:
That's not how downgrade need to apply, downgrade must not introduce feature that not exists before

Reply:
There exists a precedent for using one rector from another:

https://github.com/rectorphp/rector-src/blob/34c61ba4498150972db843fca807416540ca71e8/rules/DowngradePhp70/Rector/String_/DowngradeGeneratedScalarTypesRector.php#L45-L51

Do you think calling `DowngradePipeToMultiCatchExceptionRector` internally would be a good idea? Or should I just copy the `DowngradePipeToMultiCatchExceptionRector` code here?

Reply:
I think you can extract the logic

https://github.com/rectorphp/rector-src/blob/34c61ba4498150972db843fca807416540ca71e8/rules/DowngradePhp71/Rector/TryCatch/DowngradePipeToMultiCatchExceptionRector.php#L60-L84

into separate service, eg: `PipedExceptionManipulator` to eg: to `DowngradePhp71\NodeManipulator` namespace so can be re-used here

Reply:
It turns out I do not actually need any logic, everything is already handled by `NodesToAddCollector`.